src/NzbDrone.Core/Blacklisting/BlacklistService.cs

```diff
@@ -45,12 +54,48 @@ public void Delete(int id)
             _blacklistRepository.Delete(id);
         }
 
-        private static bool HasSamePublishedDate(Blacklist item, DateTime publishedDate)
+        private bool SameNzb(Blacklist item, ReleaseInfo release)
+        {
+            if (item.PublishedDate == release.PublishDate)
+            {
+                return true;
+            }
+
+            if (!HasSameIndexer(item, release.Indexer) &&
+                HasSamePublishedDate(item, release.PublishDate) &&
+                HasSameSize(item, release.Size))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private bool HasSameIndexer(Blacklist item, string indexer)
+        {
+            if (item.Indexer.IsNullOrWhiteSpace())
+            {
+                return true;
+            }
+
+            return item.Indexer.Equals(indexer, StringComparison.InvariantCultureIgnoreCase);
+        }
+
+        private bool HasSamePublishedDate(Blacklist item, DateTime publishedDate)
         {
             if (!item.PublishedDate.HasValue) return true;
 
-            return item.PublishedDate.Value.AddDays(-2) <= publishedDate &&
-                   item.PublishedDate.Value.AddDays(2) >= publishedDate;
+            return item.PublishedDate.Value.AddMinutes(-2) <= publishedDate &&
+                   item.PublishedDate.Value.AddMinutes(2) >= publishedDate;
```

Comment:
I've seen releases from different indexers that are even within 2 minutes that are very likely different releases, whereas the same release is likely within 10 seconds of it.

This definitely goes the other direction from where we were, allowing downloads from the same indexer with the same name and a different post times, but still allowing some grace period on different indexers.